server/graph/resolvers/asset.js

```diff
@@ -41,6 +41,25 @@ module.exports = {
         const path = parentPath ? `${parentPath}/${r.slug}` : r.slug
         return WIKI.auth.checkAccess(context.req.user, ['read:assets'], { path })
       })
+    },
+    async folderByPath(obj, args, context) {
+      const parts = args.path.toLowerCase().split('/').filter(Boolean)
+      let parentId = null
+      for (const slug of parts) {
+        const folder = await WIKI.models.assetFolders.query().where({
+          parentId: parentId,
+          slug: slug
+        }).first()
```

Comment:
Avoid querying the DB on every path segment. There's already a method to get the hierarchy in 1 query:
```
const folderHierarchy = await WIKI.models.assetFolders.getHierarchy(args.folderId)
```
Your code should either use it or do something similar.

Reply:
Thanks! Would a single assetFolders.query() be acceptable? I'd like to use getHierarchy(), but it requires a folderId.

Reply:
You're fetching all folders into memory, which isn't better. See how the getHierarchy() method works. It makes a single query using the withRecursive() method, fetching only the necessary data.

Reply:
Got it. Thanks!

Reply:
Refactored to use recursive SQL similar to getHierarchy(), starting from root and iterating downward instead of up from child/leaf. Thanks for your suggestion. Tested with PostgreSQL 15 and SQL Server 2022 (16).